### PR TITLE
View all grades

### PR DESCRIPTION
**Pull request by tim-ahn**
Screen Shot 2020-07-17 at 10 47 04 AM

updated Grade and GradeTable class components

### CODE REVIEW

File: client/components/Grade.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      grade: {
+        name: '',
+        course: '',
+        grade: ''
+      }
+    };
+  }
```
Comment (thebearingedge) [inline client/components/Grade.jsx:13]: This component doesn't need a constructor or state.

File: client/components/app.jsx [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      grades: []
+    };
+  }
+
+  componentDidMount() {
+
+  }
```
Comment (thebearingedge) [inline client/components/app.jsx:15]: This should fetch the grades from the server.

File: client/components/gradeTable.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    return (
+      <table className="table">
+        <tr>
+          <td className="header-row">Student Name</td>
+          <td className="header-row">Course</td>
+          <td className="header-row">Grade</td>
+        </tr>
+        <Grade />
+        <Grade />
+        <Grade />
```
Comment (thebearingedge) [inline client/components/gradeTable.jsx:22]: You should `.map()` through `this.props.grades` to generate the table rows.

File: client/components/gradeTable.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import Grade from './Grade';
+
+class GradeTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      grades: []
+    };
+  }
```
Comment (thebearingedge) [inline client/components/gradeTable.jsx:10]: This component does not need a constructor or state.

File: server/public/styles.css [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  padding-bottom: 3px;
+  height: 3rem;
+  background-color: lightgray;
+}
+.row {
+  height: 2rem;
+}
+tr:nth-child(2n) {
+  background-color: whitesmoke;
+}
```
Comment (thebearingedge) [inline server/public/styles.css:19]: Bootstrap has classes to style tables already.

File: server/public/styles.css [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+.header-row {
+  padding-top: 3px;
+  padding-bottom: 3px;
+  height: 3rem;
+  background-color: lightgray;
+}
+.row {
+  height: 2rem;
+}
```
Comment (thebearingedge) [inline server/public/styles.css:16]: Bootstrap already defines this `.row` class. You should not add any styles to it.

**Comment by tim-ahn** [comment]
Screen Shot 2020-07-17 at 4 24 38 PM

File: client/components/gradeTable.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+function GradeTable(props) {
+  const element = props.grades.map(grades => (
+    <tr key={grades.id}>
+      <td>{grades.name}</td>
+      <td>{grades.course}</td>
+      <td>{grades.grade}</td>
+    </tr>
+  )
```
Comment (thebearingedge) [inline client/components/gradeTable.jsx:10]: You should import your `Grade` component into this file and use that to make the table rows.

File: client/components/gradeTable.jsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Grade from './Grade';
+
+function GradeTable(props) {
+  const element = props.grades.map(grades => Grade(grades));
```
Comment (thebearingedge) [inline client/components/gradeTable.jsx:5]: Ok, we are so dang close. You want a React element of type `Grade`

```jsx
<Grade key={grade.id} name={grade.name} etc etc />
```

**Pull request merged**